android/src/main/java/com/brentvatne/common/api/Source.kt

```diff
@@ -0,0 +1,205 @@
+package com.brentvatne.common.api
+
+import android.annotation.SuppressLint
+import android.content.ContentResolver
+import android.content.Context
+import android.content.res.Resources
+import android.net.Uri
+import android.text.TextUtils
+import com.brentvatne.common.toolbox.DebugLog
+import com.brentvatne.common.toolbox.DebugLog.e
+import com.brentvatne.common.toolbox.ReactBridgeUtils.safeGetArray
+import com.brentvatne.common.toolbox.ReactBridgeUtils.safeGetInt
+import com.brentvatne.common.toolbox.ReactBridgeUtils.safeGetMap
+import com.brentvatne.common.toolbox.ReactBridgeUtils.safeGetString
+import com.facebook.react.bridge.ReadableMap
+import java.util.Locale
+
+/**
+ * Class representing Source props for host.
+ * Only generic code here, no reference to the player.
+ */
+class Source {
+    /** String value of source to playback */
+    private var uriString: String? = null
+
+    /** Parsed value of source to playback */
+    var uri: Uri? = null
+
+    /** Start position of playback used to resume playback */
+    var startPositionMs: Int = -1
+
+    /** Will crop content start at specified position */
+    var cropStartMs: Int = -1
+
+    /** Will crop content end at specified position */
+    var cropEndMs: Int = -1
+
+    /** Allow to force stream content, necessary when uri doesn't contain content type (.mlp4, .m3u, ...) */
+    var extension: String? = null
+
+    /** Metadata to display in notification */
+    var metadata: Metadata? = null
+
+    /** http header list */
+    val headers: MutableMap<String, String> = HashMap()
+
+    /** return true if this and src are equals  */
+    fun isEqual(src: Source?): Boolean {
+        if (src == null) return false
+        return (
+            uri == src.uri &&
+                cropStartMs == src.cropStartMs &&
+                cropEndMs == src.cropEndMs &&
+                startPositionMs == src.startPositionMs &&
+                extension == src.extension
+            )
+    }
```

Comment:
nit;
How about override `Object.equals`? if you override it, you can use "==" within Kotlin

Reply:
Good idea, I updated the code to override ==, thank you

Reply:
fyi) If you override equals "==" works in Kotlin, but not in Java.

Reply:
Thanks for the reminder, let's forget java 😂

Reply:
oups, you are right, I forgot we still use java 🙄